### ISSUE TITLE
Support for automatic REST-JSON endpoint for grpc classes

### ISSUE THREAD

**Issue by krickert** [micronaut-projects/micronaut-grpc#995]
### Feature description

By combining Micronaut’s introspection and dynamic routing capabilities with the `jackson-datatype-protobuf` library, you can write a single generic mechanism that exposes any gRPC service as a JSON-based REST endpoint. Mark your gRPC services with a special annotation, and at startup, automatically build REST endpoints. JSON input is transformed into Protobuf messages and vice versa without needing separate code generation or schema conversion steps.

https://github.com/HubSpot/jackson-datatype-protobuf

Would be used to handle the serialization.

Approach --
**Annotate gRPC Service with a Custom Annotation**
Define a custom annotation, for example @RestExposed, and place it on your gRPC service implementations. This will indicate that the service should also be exposed as a REST endpoint.

```
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class RestExposed

```
Then we use it

```
@RestExposed
@Singleton
class MyGrpcService : MyServiceGrpc.MyServiceImplBase() {
    override fun getSomething(request: MyRequest, responseObserver: StreamObserver<MyResponse>) {
        // gRPC logic...
    }
}
```
Then register the protobuf serializer with jackson 

```
@Factory
class ObjectMapperFactory {
    @Singleton
    fun objectMapper(): ObjectMapper {
        return ObjectMapper()
            .registerModule(com.hubspot.jackson.datatype.protobuf.ProtobufModule())
    }
}
```

Basically, if you create a new grpc service, we can dynamically make the service exposed as a REST service

Does this idea sound useful?

**Comment by krickert** [comment]
Service created - will see about how to turn this into a feature

https://gitlab.com/ragsearch/pipelines/grpc-rest-exposed
